roles/v2/pool/src/status.rs

```diff
@@ -0,0 +1,115 @@
+use crate::error::PoolError;
+
+/// Each sending side of the status channel 
+/// should be wrapped with this enum to allow 
+/// the main thread to know which component sent the message
+#[derive(Debug)]
+pub enum Sender {
+    Downstream(async_channel::Sender<Status>),
+    DownstreamListener(async_channel::Sender<Status>),
+    Upstream(async_channel::Sender<Status>),
+}
+
+impl Sender {
+    /// used to clone the sending side of the status channel used by the TCP Listener
+    /// into individual Sender's for each Downstream instance
+    pub fn listener_to_connection(&self) -> Self {
+        match self {
+            Self::DownstreamListener(inner) => Self::Downstream(inner.clone()),
+            _ => unreachable!(),
+        }
+    }
+
+    pub async fn send(&self, status: Status) -> Result<(), async_channel::SendError<Status>> {
```

Comment:
This is currently never used so can you add a `_` to it - I'd love to get to inbox(0) on these types of warnings.